apisix/ssl.lua

```diff
@@ -125,21 +114,18 @@ end
 
 function _M.aes_encrypt_pkey(origin, field)
     local local_conf = core.config.local_conf()
+    local aes_128_cbc_with_iv_tbl_gde = get_aes_128_cbc_with_iv_gde(local_conf)
+    local aes_128_cbc_with_iv_gde = aes_128_cbc_with_iv_tbl_gde[1]
 
-    if not field then
-        -- default used by ssl
-        local aes_128_cbc_with_iv_tbl_ssl = get_aes_128_cbc_with_iv_ssl(local_conf)
-        local aes_128_cbc_with_iv_ssl = aes_128_cbc_with_iv_tbl_ssl[1]
-        if aes_128_cbc_with_iv_ssl ~= nil and core.string.has_prefix(origin, "---") then
-            return encrypt(aes_128_cbc_with_iv_ssl, origin)
-        end
+    if aes_128_cbc_with_iv_gde == nil then
+        return origin
+    end
+
+    if not field and core.string.has_prefix(origin, "---") then
```

Comment:
Do we still need the `field` parameter?

Reply:
I reserved this parameter because ssl encryption is need to check whether the `origin` has a `---` prefix and data  encryption is not. If I remove it, is the prefix check of `origin` should be also removed too? or just reserve the check and put it to ssl and data encryption?

Reply:
I prefer the later one: `reserve the check and put it to ssl and data encryption`, the `data encryption` don't have a prefix like `---`, so no need to consider the `data encryption`